src/client.ts

```diff
@@ -255,10 +255,14 @@ export default class Client {
   public async deposit(params: PostDepositRequestParams): Promise<GetBalanceResponse> {
     const client = this.getClient();
     const assetId = params.assetId || AddressZero;
-    if (params.assetId === AddressZero) {
-      delete params.assetId;
+    const response = await client.deposit({
+      amount: params.amount,
+      assetId: params.assetId || undefined,
+    });
+    
+    if (params.requestCollateral && params.assetId) {
+      client.requestCollateral(params.assetId).catch(this._logger.error);
```

Comment:
Should we also wait for the collateral request to finish before returning the response?

Reply:
I'm thinking it's better not to, partly based on the goal described here https://github.com/ExchangeUnion/xud/issues/1753 to make the deposit request return sooner - as soon as the channel is open. But @kilrau can weigh in if he thinks otherwise.

Reply:
Hmm well we need to know if our collateral request was received and accepted by the node. So if the call returns directly after the node accepted our request (milliseconds), we wait for the call. If not, that behavior should probably be changed on connext side.

Reply:
Note that it's not possible to parallelize depositing/requesting collateral currently because of how sending funds into the channel works.

What's the verdict here? We're trying to move towards a pattern where we just try to resolve promises quickly and give you wrapped tx objects where you can separately just call `await tx.wait()` without blocking the whole process until tx gets mined (e.g. with your CLI).

If we're doing the above, then I wouldn't recommend making both the deposit/request collateral in a single call, as you would then be *forced* to wait for the deposit to be mined when making the deposit call.

Does thtat make sense?

Reply:
So the flow would be `/deposit` and `await tx.wait()` as soon as that returns call `client.requestCollateral`. All without blocking the CLI. Correct?

Reply:
@ArjunBhuptani Are you suggesting then to make a separate endpoint for collateral requests only? So we would call `/deposit` followed by `/request-collateral` or similar.

Also, I see how we could await a promise within this rest-api-client, but for the service communicating over http/rest (xud, in our case) we wouldn't have access to that promise directly. So if we wanted to act upon both the beginning of the tx and upon the tx completing, I figure we'd need an endpoint for the rest-client to send a request to upon transactions completing, right ?

But for the purposes here, if I understand correctly we would be fine just calling `/deposit`, completing the cli call, and then calling `/request-collateral` without the cli needing to wait for that to complete. I'll add more xud-specific thoughts in the related issue.

Reply:
@sangaman, we can definitely add the `request-collateral` method. This is already one of our top-level client methods, so it makes sense to expose it as a REST API method as well. We will make sure this is included in the next release.

> Also, I see how we could await a promise within this rest-api-client, but for the service communicating over http/rest (xud, in our case) we wouldn't have access to that promise directly. So if we wanted to act upon both the beginning of the tx and upon the tx completing, I figure we'd need an endpoint for the rest-client to send a request to upon transactions completing, right ?

One thing we could do is return the tx hash of the submitted tx which allows you to check the status on your own using an ETH provider.

Reply:
Thanks. I'll put in a `request-collateral` endpoint then, seems like a sensible approach to me.